packages/server/modules/workspaces/services/workspaceLimits.ts

```diff
@@ -0,0 +1,31 @@
+import { getPaginatedProjectModelsTotalCountFactory } from '@/modules/core/repositories/branches'
+import { getProjectDbClient } from '@/modules/multiregion/utils/dbSelector'
+import {
+  GetWorkspaceModelCount,
+  QueryAllWorkspaceProjects
+} from '@/modules/workspaces/domain/operations'
+
+// TODO: What is correct DI for a service that has to dynamically select a regional database?
```

Comment:
Inject getPaginatedProjectModelsTotalCount too, but when you feed it in, add project DB resolution code

```
getWorkspaceModelCountFactory({
  getPaginatedProjectModelsTotalCount: async (projectId) => {
    const regionDb = await getProjectDbClient({ projectId })
    const getPaginatedProjectModelsTotalCount = getPaginatedProjectModelsTotalCountFactory({db: regionDb})
    return getPaginatedProjectModelsTotalCount({..})
  }
})
```

Reply:
sick, done